infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/ClusterDescriptor.java

```diff
@@ -16,67 +14,36 @@ public class ClusterDescriptor {
     private static int CORFU_PORT = 9000;
 
     @Getter
-    String clusterId;
+    final String clusterId;
 
     @Getter
-    ClusterRole role;
+    final ClusterRole role;
 
     @Getter
-    List<NodeDescriptor> nodesDescriptors;
+    final List<NodeDescriptor> nodeDescriptors;
 
     @Getter
-    private int corfuPort;    // Port on which Corfu DB runs on this cluster
+    final private int corfuPort;    // Port on which Corfu DB runs on this cluster
 
-    public ClusterDescriptor(String clusterId) {
-        this.clusterId = clusterId;
-    }
-
-    public ClusterDescriptor(ClusterConfigurationMsg clusterConfig) {
-        this.clusterId = clusterConfig.getId();
-        this.role = clusterConfig.getRole();
-        this.corfuPort = clusterConfig.getCorfuPort() != 0 ? clusterConfig.getCorfuPort() : CORFU_PORT;
-        this.nodesDescriptors = new ArrayList<>();
-        for (NodeConfigurationMsg nodeConfig : clusterConfig.getNodeInfoList()) {
-            NodeDescriptor newNode = new NodeDescriptor(nodeConfig.getAddress(),
-                    Integer.toString(nodeConfig.getPort()), clusterId,
-                    nodeConfig.getConnectionId(),
-                    nodeConfig.getNodeId());
-            this.nodesDescriptors.add(newNode);
-        }
-    }
 
     public ClusterDescriptor(ClusterDescriptor info, ClusterRole roleType) {
         this.clusterId = info.clusterId;
         this.role = roleType;
-        this.nodesDescriptors = new ArrayList<>();
+        this.nodeDescriptors = new ArrayList<>();
```

Comment:
the role is used for federation usecase only to determine if the role of local cluster has changed.
I thought we can retain "role" for that. Otherwise we'll need to check if any remote cluster moved buckets (i.e.,. from remoteSink bucket to remoteSource bucket or vice versa)..more lines of code when we can just use "role".

And another use of "role" in the code is in the defaultClusterManger, to simulate different kinds of topology. This can be changed to use the clusterIds as well if we want though.

Reply:
discussed offline and removed "role"